File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> numsMap;
        int n = nums.size();

        for (int i = 0; i < n; i++) {
            if (numsMap.count(nums[i])) {
                return true;
            }
            else {
                numsMap[nums[i]] = i;
            }
        }

        return false;
    }
};

File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-0.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        def backtrack(i, s):
            if i >= len(nums):
                return s
            return backtrack(i + 1, s ^ nums[i]) + backtrack(i + 1, s)
        return backtrack(0, 0)